src/test/java/org/candlepin/controller/ConsumerManagerTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCloudData;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.service.EventAdapter;
+import org.candlepin.service.model.CloudCheckInEvent;
+import org.candlepin.test.TestUtil;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Date;
+
+@ExtendWith(MockitoExtension.class)
+public class ConsumerManagerTest {
```

Comment:
In general, I'm not a huge fan of the way mocks are leveraged for this test suite. For the curator and adapter it's borderline, but because the mocks aren't complete, they're very brittle and aren't going to be representative of any actual implementation. For data or model objects, those should never be mocked as (a) it's more work to mock them out than just using a real object directly, and (b) you're highly likely to violate some of the target's invariants by failing to capture all of its behavior without mocking out the whole class/graph, which loops back into (a).

I can be convinced that mocks of other units can be used with minimal impact on long-term maintenance while staying true to the actual implementations we have, but I have yet to see it done well. Verifications only tell us so much, and are heavily tied to a specific implementation of the unit under test -- not necessarily that the actual behavior of the UUT is correct.

Reply:
ack